src/sinks/elasticsearch/common.rs

```diff
@@ -1,6 +1,8 @@
 use std::collections::HashMap;
 
+#[cfg(feature = "aws-core")]
 use aws_credential_types::provider::SharedCredentialsProvider;
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏ Is this still needed? If it is, the feature flagged use statement could conceivably be removed from the top of the file here and the type or trait path be inlined where needed, or if in a block, the use statement can be moved there.

```suggestion
```

Reply:
hi @neuronull thx for you advice. I modified all you mentioned code.